include/targets/hip/quda_fp16.cuh

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <hip/hip_fp16.h>
+
+namespace quda
```

Comment:
The habs2 code is duplicated. Worth avoiding that here?

Reply:
I think leave this as it is for now.  When we flick the switch and require CUDA 11 for QUDA on NVIDIA, the implementation for CUDA won't be needed since it's present in the toolkit.  At that point we can change the CUDA variant to just
```c++
  __device__ inline half2 habs2(half2 input) { return __habs2(input); }
  ```